libs/common/src/lib/tasks/dto/create-tasks.dto.ts

```diff
@@ -11,4 +11,6 @@ export class CreateTaskDto {
   readonly arguments?: string[];
   readonly options?: object;
   readonly priority?: TaskPriority;
+  readonly priorityIndex?: number;
```

Comment:
Priority index could be used to give higher priority to tasks that have more dependent on them tasks.
For example: 
LibA, LibB, and LibC depend on LibD, so LibD has 3 dependent tasks. priorityIndex in this case will be 3.
LibA, LibB depend on LibE, so LibE has 2 dependent tasks. priorityIndex in this case will be 2.

So, from the example, we can clearly see that building LibD first unblocks LibC and it could be picked up right away by available runner.